nupic/research/frameworks/dynamic_sparse/models/loggers.py

```diff
@@ -0,0 +1,220 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from collections import defaultdict
+
+import numpy as np
+import torch
+from pandas import DataFrame
+
+
+class BaseLogger:
+    def __init__(self, model, config=None):
+        defaults = dict(debug_weights=False, verbose=0)
+        defaults.update(config or {})
+        self.__dict__.update(defaults)
```

Comment:
Small note: I've thought about the method of setting class attributes via the config a bit, and there's potential edge case which would should avoid - overwriting pre-existing attributes. As well, I think we should avoid adding more attributes than necessary. If config has more params than just `debug_weights` and `verbose`, we'd not only update those attrs, but we also add other params and potentially change the value of some pre-existing keys in `self.__dict__`. 

Something like the following would work
```
        config = config or {}
        defaults = dict(
            debug_weights=False,
            verbose=0,
        )
       # Only update desired defaults. 
        new_defaults = {k: (config.get(k, None) or v) for k, v in defaults.items()}
        self.__dict__.update(new_defaults)
```

Reply:
It is implemented that way specifically to override attributes (e.g. a new class can set a new default to a variable, different than the parent class). But I did not think through this a lot, this was the quick and dirty solution, we can come up with something better if that is not working.